src/plugins/ui_feature/find_and_replace.ts

```diff
@@ -216,45 +329,57 @@ export class FindAndReplacePlugin extends UIPlugin {
   /**
    * Replace the value of the currently selected match
    */
-  private replace(replaceWith: string) {
-    if (this.selectedMatchIndex === null || !this.currentSearchRegex) {
+  private replace(replaceWith: string, updateSearch: boolean = true) {
+    const selectedMatch =
+      this.selectedMatchIndex !== null ? this.searchMatches[this.selectedMatchIndex] : null;
+    if (!selectedMatch || !this.currentSearchRegex) {
       return;
     }
-    const matches = this.searchMatches;
-    const selectedMatch = matches[this.selectedMatchIndex];
-    const sheetId = this.getters.getActiveSheetId();
-    const cell = this.getters.getCell({ sheetId, ...selectedMatch });
+    const cell = this.getters.getCell(selectedMatch);
     if (cell?.isFormula && !this.searchOptions.searchFormulas) {
-      this.selectNextCell(Direction.next);
+      if (updateSearch) {
+        this.selectNextCell(Direction.next);
+      }
     } else {
```

Comment:
do eary return rather than if/else.  And add empty lines sometimes. The whole function is kinda hard to read